src/main/java/org/openstreetmap/atlas/geography/atlas/command/AtlasFindByAtlasIdentifierSubCommand.java

```diff
@@ -0,0 +1,125 @@
+            Command.Optionality.REQUIRED);
+
+    private static final Command.Switch<String> JOINED_OUTPUT_PARAMETER = new Command.Switch<>(
+            "joinedOutput",
+            "The Atlas file to save the joined output to (optional). If not passed the found shards will not be joined and only appear in the console.",
+            String::toString, Command.Optionality.OPTIONAL);
+
+    private final Set<Long> ids = new HashSet<>();
+
+    private final Set<String> shardNames = new HashSet<>();
+
+    public AtlasFindByAtlasIdentifierSubCommand()
+    {
+        super("find-atlas-id",
+                "Find which atlas files contain particular Atlas features using a given set of Atlas identifiers");
+    }
+
+    @Override
+    public Command.SwitchList switches()
+    {
+        return super.switches().with(ATLAS_ID_PARAMETER, JOINED_OUTPUT_PARAMETER);
+    }
+
+    @Override
+    public void usage(final PrintStream writer)
+    {
+        writer.printf(AtlasCommandConstants.INPUT_PARAMETER_DESCRIPTION);
+        writer.printf("-id=1000000,2000000 : comma separated Atlas identifiers to search for\n");
+        writer.printf("-joinedOutput=path/to/joined.atlas : the path to the output Atlas file\n");
+    }
+
+    @Override
+    protected void start(final CommandMap command)
+    {
+        // Collect ids
+        this.ids.addAll((Set) command.get(ATLAS_ID_PARAMETER));
+    }
+
+    @Override
+    protected void handle(final Atlas atlas, final CommandMap command)
+    {
+        // Get all atlas entities with ids matching the input list
+        atlas.entities(idCheck()).forEach(item ->
+        {
+            // Print atlas and item information
+            System.out.printf(formatAtlasObject(atlas, item));
+            // Record shard name
+            this.shardNames.add(atlas.getName());
+        });
+    }
+
+    @Override
+    protected int finish(final CommandMap command)
+    {
+        final Optional output = command.getOption(JOINED_OUTPUT_PARAMETER);
+        // If joining is requested and there are shards to join...
+        if (output.isPresent() && !this.shardNames.isEmpty())
+        {
+            System.out.printf("Joining...");
+            // Use AtlasJoinerSubCommand to join found atlases
+            AtlasReader.main("join",
+                    String.format("-input=%s", command.get(super.switches().get(0))),
+                    String.format("-output=%s", output.get()),
+                    String.format("-atlases=%s", String.join(",", this.shardNames)));
+        }
+        return 0;
+    }
+
+    /**
+     * Predicate to check an {@link AtlasObject} against the list of ids.
+     *
+     * @param <T>
+     *            Object type
+     * @return {@link Predicate}
+     */
+    private <T extends AtlasObject> Predicate<T> idCheck()
+    {
+        return object -> this.ids.contains(((Long) object.getIdentifier()).toString());
```

Comment:
Why do we need to cast as `Long` and do a `toString()`? Why not just see if the `Set` of type `Long` contains the object identifier?